RELEASE/scripts/autoscend/auto_restore.ash

```diff
@@ -0,0 +1,991 @@
+		{
+			equip(replace);
+		}
+		cli_execute("outfit Backup");
+		if(closet)
+		{
+			if(item_amount(grab) > 0)
+			{
+				put_closet(1, grab);
+			}
+		}
+
+	}
+	else
+	{
+		set_property("restUsingChateau", false);
+		cli_execute("rest");
+		set_property("restUsingChateau", true);
+	}
+	return get_property("timesRested").to_int();
+}
+
+boolean haveFreeRestAvailable(){
+	return get_property("timesRested").to_int() < total_free_rests();
+}
+
+int freeRestsRemaining(){
+	return max(0, total_free_rests() - get_property("timesRested").to_int());
+}
+
+boolean haveAnyIotmAlternateCampsight(){
+	return chateaumantegna_available() || auto_campawayAvailable();
+}
+
+/*
+ * Try to use a free rest.
+ *
+ * returns true if a rest was used false if it wasnt (for any reason)
+ */
+boolean doFreeRest(){
+	if(haveFreeRestAvailable()){
+		int rest_count = get_property("timesRested").to_int();
+		return doRest() > rest_count;
+	}
+	return false;
+}
+
+float mp_regen()
+{
+	return 0.5 * (numeric_modifier("MP Regen Min") + numeric_modifier("MP Regen Max"));
+}
+
+float hp_regen()
+{
+	return 0.5 * (numeric_modifier("HP Regen Min") + numeric_modifier("HP Regen Max"));
+}
+
+boolean uneffect(effect toRemove)
+{
+	if(have_effect(toRemove) == 0)
+	{
+		return true;
+	}
+	if(($effects[Driving Intimidatingly, Driving Obnoxiously, Driving Observantly, Driving Quickly, Driving Recklessly, Driving Safely, Driving Stealthily, Driving Wastefully, Driving Waterproofly] contains toRemove) && (auto_get_campground() contains $item[Asdon Martin Keyfob]))
+	{
+		string temp = visit_url("campground.php?pwd=&preaction=undrive");
+		return true;
+	}
+
+	if(cli_execute("uneffect " + toRemove))
+	{
+		//Either we don\'t have the effect or it is shruggable.
+		return true;
+	}
+
+	if(item_amount($item[Soft Green Echo Eyedrop Antidote]) > 0)
+	{
+		visit_url("uneffect.php?pwd=&using=Yep.&whicheffect=" + to_int(toRemove));
+		print("Effect removed by Soft Green Echo Eyedrop Antidote.", "blue");
+		return true;
+	}
+	else if(item_amount($item[Ancient Cure-All]) > 0)
+	{
+		visit_url("uneffect.php?pwd=&using=Yep.&whicheffect=" + to_int(toRemove));
+		print("Effect removed by Ancient Cure-All.", "blue");
+		return true;
+	}
+	return false;
+}
+
+// Deprecated, please use acquireHP()
+boolean useCocoon()
+{
+  return acquireHP();
+}
+
+/**
+ * we could in theory set this as our restore script, but autoscend is not currently designed to heal this way and changing this now would probably break assumptions people have anticipated in their code, causing undefined behavior. I assume this is why we have the warning about autoscend not playing well with restore scripts and disabling them when it starts.
+ *
+ * Additionally this script would require some number of imports of other methods (mostly auto_util.ash) which may or may not be easy to do. I did try once by just importing autoscend, but I ended up with an infinite loop. So for now this is just here for posterity.
```

Comment:
I've absolutely created wrapper scripts that `import <sl_ascend.ash>` at the top - it's useful when I need a one-shot script for Hobopolis and I want to use functions in sl_util. So I think this would be very possible.

I agree we shouldn't do it. Just wanted to point out that it's possible. I'm not sure about the infinite loop.

Reply:
I'm sure it is, i was probably just doing something silly. Im fairly confident mafia is smart enough to work out the imports, but i tried a few times to call the script directly and it just spun out and forced me to restart mafia before i could do anything. Problem for another day.